### DISCUSSION TITLE
How to store and restore cookies for multiple browser instances

### DISCUSSION THREAD

**Discussion by mike55554** [cefsharp/CefSharp#3845] [General]
Hello,
I'm trying to implement multiple browser instances in one application. On startup they should load the cookies and their cache from the previous run of the application. 

ChromiumWebBrowser w1 = new ChromiumWebBrowser("www.google.de");
ChromiumWebBrowser w2 = new ChromiumWebBrowser("www.google.de");

w1.RequestContext = new RequestContext(new RequestContextSettings { CachePath = cache + "1", PersistSessionCookies = true });
w2.RequestContext = new RequestContext(new RequestContextSettings { CachePath = cache + "2", PersistSessionCookies = true });

Browser w1 and w2 is isolated, but it don't reuse the cache from a previous session.

How can i fix this?

**Comment by amaitland** [comment]
In future please use discussions to ask questions. https://github.com/cefsharp/CefSharp/discussions

>  RequestContextSettings.CachePath must be equal to or a child of CefSettings.RootCachePath see https://github.com/cefsharp/CefSharp/issues/3111#issuecomment-629713608

As per https://github.com/cefsharp/CefSharp/wiki/General-Usage#request-context-browser-isolation you need to set CefSettings.RootCachePath
